packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart

```diff
@@ -129,6 +130,15 @@ mixin ShapeHitbox on ShapeComponent implements Hitbox<ShapeHitbox> {
     return collisionAllowed && aabb.intersectsWithAabb2(other.aabb);
   }
 
+  /// Returns the reflection of the ray against the closest.
+  ///
+  /// If you are only interested in the intersection point use
+  /// [RaycastResult.intersectionPoint] of the result.
+  RaycastResult<ShapeHitbox>? rayIntersection(
+    Ray2 ray, {
+    RaycastResult<ShapeHitbox>? out,
+  });
```

Comment:
It would be good to start moving towards using `Shape` classes more, instead of putting more geometric operations into shape-hitbox components.

Reply:
For sure, but the `Shape` classes doesn't have the correct projection yet right?